arangod/RocksDBEngine/RocksDBCuckooIndexEstimator.h

```diff
@@ -337,14 +337,14 @@ class RocksDBCuckooIndexEstimator {
 
   double computeEstimate() {
     READ_LOCKER(locker, _lock);
-    if (0 == (_nrTotal + _nrCuckood)) {
-      TRI_ASSERT(0 == _nrTotal);
-      TRI_ASSERT(0 == _nrCuckood);
+    if (0 == _nrTotal) {
+      TRI_ASSERT(0 == _nrUsed);
       // If we do not have any documents we have a rather constant estimate.
       return 1.0;
     }
-    return (static_cast<double>(_nrUsed) /
-            static_cast<double>(_nrTotal + _nrCuckood));
+    TRI_ASSERT(_nrUsed <= _nrTotal);
+
```

Comment:
Should we instead of ASSERT use `if (_nrUsed >= _nrTotal + _nrCuckood) { _nrUsed = _nrTotal + _nrCuckood; }
This counter is an estimate and as we do not exactly know how many elements we have cuckood out the nrUsed may actually be larger than the other two values.

Reply:
With the changes, `_nrUsed` should always be at most `_nrTotal` (as an invariant). If that's not true, we should probably detect it via assert in maintainer mode, right?

Reply:
Yes i think you are right. we should include the ASSERT for us. but the "repair" for production :+1: